tictactoe/dsl/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/dsl/TicTacToeVerify.kt

```diff
@@ -0,0 +1,44 @@
+package com.anaplan.engineering.azuki.tictactoe.dsl
+
+import com.anaplan.engineering.azuki.core.dsl.Verify
+import com.anaplan.engineering.azuki.core.scenario.ScenarioQueries
+import com.anaplan.engineering.azuki.core.system.DerivedQuery
+import com.anaplan.engineering.azuki.core.system.Query
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.Position
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.TicTacToeQueryFactory
+
+class TicTacToeVerify(private val queryFactory: TicTacToeQueryFactory) : Verify<TicTacToeQueryFactory> {
+
+    fun <T, C : Collection<T>> forAll(
+        getDriver: DerivedQueryBlock.() -> (qf: TicTacToeQueryFactory) -> Query<C>, v: TicTacToeVerify.(T) -> Unit
+    ) {
+        derivedQueriesList.add(queryFactory.createForAllQuery(DerivedQueryBlock().getDriver()) { t: T, qf: TicTacToeQueryFactory ->
+            val verify = TicTacToeVerify(qf).apply { v(t) }
+
+            // TODO: relax these requirements by creating a derived query concatenation operator?
+            require(verify.queriesList.isEmpty() || verify.derivedQueriesList.isEmpty()) { "forAll cannot have both derived and non-derived queries" }
+            require(verify.derivedQueriesList.size <= 1) { "forAll cannot have more than one nested derived query" }
```

Comment:
This is a slightly obnoxious restriction as it's not obvious syntactically that it's in place.

Like I said in the comment here, I suspect that it could be possible to fix this by creating some sort of operator that chains together multiple derived queries of the same type into one derived query.  Of course, it may well be that (to quote a famous fictitious scientist) I'm spending so much time thinking about whether I can that I haven't stopped to think about whether I should.